File: 1833-find-the-highest-altitude/find-the-highest-altitude.java
class Solution {
     public int largestAltitude(int[] gain) {
          int n = gain.length;
          int pre=0;
          int max = Integer.MIN_VALUE;
          for(int i=0;i<n+1;i++){
             max=Math.max(max,pre);
             if(i<n) pre+=gain[i];
           }
           return max;
     }
}

File: 1833-find-the-highest-altitude/find-the-highest-altitude.go
 package main
 import (
 "fmt"
 "math"
 )
 func largestAltitude(gain []int) int {
      //TC: O(n)
      //SC: O(n)
      // golang
     cpy := make([]int,1)
     cpy[0] = 0
     for _ , ele := range gain {
     cpy = append(cpy,ele)
     }
     fmt.Println(cpy)
     ans := make([]int,1)
     ans[0] = 0
     var putele = 0
     for _,  ele := range cpy{
       putele += ele
       ans = append(ans,putele)
     }
     maxInteger := math.MinInt
     for _, ele := range ans {
        maxInteger = max(maxInteger,ele)
     }
     return maxInteger
}